sdk/include/opentelemetry/sdk/trace/span_data.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include <chrono>
+#include "opentelemetry/core/timestamp.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/sdk/trace/recordable.h"
+#include "opentelemetry/trace/canonical_code.h"
+#include "opentelemetry/trace/span_id.h"
+#include "opentelemetry/trace/trace_id.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+/**
+ * SpanData is a representation of all data collected by a span.
+ */
+class SpanData final : public Recordable
+{
+public:
+  /**
+   * Initialized all span fields with default values. Time stamps and duration
+   * are initialized with 0, the status of the span is initialized with OK.
+   */
+  SpanData() noexcept : duration_(0), status_code_(opentelemetry::trace::CanonicalCode::OK) {}
```

Comment:
Nit: Can we remove the constructor and use in-class initialization?
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c48-prefer-in-class-initializers-to-member-initializers-in-constructors-for-constant-initializers